### ISSUE TITLE
Poetry fails if a local dep is missing while only defined as extra

### ISSUE THREAD

**Issue by ClementWalter** [python-poetry/poetry#3936]
# Checklist

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Mac OS 10.15.7 (19H524)
- **Poetry version**: Poetry version 1.1.5
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

```
[tool.poetry]
name = "tmp-1"
version = "0.1.0"
description = ""
authors = ["Clement Walter <clement0walter@gmail.com>"]

[tool.poetry.dependencies]
python = "^3.7"
tmp-0 = {path = "../tmp-0", optional = true, develop = true}

[tool.poetry.extras]
tmp0 = ["tmp-0"]

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

## Issue

Poetry does not manage to handle missing deps defined as extra, even though a complete `poetry.lock` exists and has been generated previously.

Workflow to generate the error:
- create an dummy project in `tmp-0`
- create `tmp-1` project. Add `tmp-0` as extra deps as above.
- `poetry env remove python && rm -rf ../tmp-0` to start over fresh
- `poetry install` fails

```
> poetry install
  ValueError

  Directory ../tmp-0 does not exist

```

**Comment by ClementWalter** [comment]
The use case is: I am working on two packages simultaneously with the second one depending on the first one. When I am developing, I want to have the `tmp-0` I am working on installed in `develop`, but elsewhere a true dep declared as any other

**Comment by Wirg** [comment]
Hi !

I am running in the same issue with optionable path to wheels (like prodi.gy ones) without  develop mode.

```toml
[tool.poetry.dependencies]
prodigy = [
    {path = "./data/prodigy_wheels/prodigy-1.10.8-cp36.cp37.cp38-cp36m.cp37m.cp38-win_amd64.whl", markers = "sys_platform == 'win32'", optional = true},
    {path = "./data/prodigy_wheels/prodigy-1.10.8-cp36.cp37.cp38-cp36m.cp37m.cp38-linux_x86_64.whl", markers = "sys_platform == 'linux'", optional = true},
    {path = "./data/prodigy_wheels/prodigy-1.10.8-cp36.cp37.cp38-cp36m.cp37m.cp38-macosx_10_14_x86_64.whl", markers = "sys_platform == 'darwin'", optional = true},
]
[tool.poetry.extras]
prodigy = ["prodigy"]
```

**Comment by Wirg** [comment]
Hi,

Bumping this as gently as possible :)

I am still blocked by this : I have the lines commented in my pyproject.toml and have to explicitely ask people to uncomment, update, and run the install in the docs.

Is this something I could help poetry on or would the fix be too complex ?

**Comment by Wirg** [comment]
Hi !

Any updates on this ? :)

**Comment by finswimmer** [comment]
Duplicate of #668
